internal/daemonset/daemonset.go

```diff
@@ -128,6 +128,27 @@ func getEnvs() []corev1.EnvVar {
 	}
 }
 
+func getTopologyEnvs() []corev1.EnvVar {
```

Comment:
you could reuse `getEnv` and simply `append` `NODE_ADDRESS` or 
```go
type NodeEnv struct {
	Name      string
	FieldPath string
}

func getWorkerEnvs(envs ...NodeEnv) []corev1.EnvVar {
	envList := []corev1.EnvVar{}

	for _, env := range envs {
		envList = append(envList, corev1.EnvVar{
			Name: env.Name,
			ValueFrom: &corev1.EnvVarSource{
				FieldRef: &corev1.ObjectFieldSelector{
					FieldPath: env.FieldPath,
				},
			},
		})
	}

	return envList
}
```

Reply:
fixed, used the first option , seemed a little bit more readable